cluster-autoscaler/cloudprovider/gke/gke_manager.go

```diff
@@ -106,11 +107,37 @@ type GkeManager interface {
 	GetProjectId() string
 	GetClusterName() string
 	GetMigTemplateNode(mig *GkeMig) (*apiv1.Node, error)
+	GetNodeLocations() []string
+}
+
+// gkeConfigurationCache is used for storing cached cluster configuration.
+type gkeConfigurationCache struct {
+	sync.Mutex
+	nodeLocations []string
+}
+
+func (cache *gkeConfigurationCache) setNodeLocations(locations []string) {
+	cache.Lock()
+	defer cache.Unlock()
+
+	cache.nodeLocations = locations
```

Comment:
I would suggest copying slice here. That way we ensure that internal array is not modified even if caller by mistake changes array which he passed as parameter to `setNodeLocations`

Reply:
Done